packages/context-editor/src/utils/links.ts

```diff
@@ -0,0 +1,43 @@
+import type { MarkType } from "prosemirror-model";
+import type { EditorState, Transaction } from "prosemirror-state";
+
+export const emailOrUriRegexBase =
+	"(?<mailto>mailto:)?(?<emailOrUri>(?:(?:(https|http|ftp)+)://)?(?:\\S+(?::\\S*)?(?<atSign>@))?(?:(?:([a-z0-9][a-z0-9-]*)?[a-z0-9]+)(?:\\.(?:[a-z0-9-])*[a-z0-9]+)*(?:\\.(?:[a-z]{2,})(:\\d{1,5})?))(?:/[^\\s]*)?)";
```

Comment:
This is just what we were using in legacy. It matches a little too frequently, since it doesn't know which TLDs are real. We could improve that (at the cost of performance?) by using a list like https://www.npmjs.com/package/tlds. We could compromise by just taking a subset of the most common TLDs too. Thoughts?

Reply:
I think we can stick with this for now, since it's easy enough to undo a link if it matches incorrectly